### PR TITLE
Finalise name/ID conventions.

### LINKED ISSUE #127: object.id/object.name convention
We currently have a convention that objects in the catalog have an ``id`` and a ``name``. The ``id`` is a unique, short, memorable identifier. The ``name`` is how you would refer to this object in an English sentence. For example, ``homsap`` is the species ID and ``Homo sapiens`` is the name. The ``name`` field is essential for building the catalog documentation in HTML and text.

Perhaps ``name`` is not quite right though, as it's not working quite so well for simulation models "Three population Out of Africa" is more like a short description...

Any thoughts?

### PR DESCRIPTION
**Pull request by jeromekelleher**
Closes #127

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Can you take a look at this and merge if you're happy? It touches a lot of the code so hard to follow up until it's merged.

**Comment by andrewkern** [comment]
yes i will review in a couple of hours... let me get these kids to school...

File: stdpopsim/catalog/arabidopsis_thaliana.py [hunk 1]
```diff
@@ -66,15 +67,16 @@
 
 _gm = stdpopsim.GeneticMap(
     species=_species,
-    name="Salome2012_TAIR7",
+    id="Salome2012_TAIR7",
+    description="Salome map FIXME",
```
Comment (andrewkern) [inline stdpopsim/catalog/arabidopsis_thaliana.py]: not sure why this says FIXME at this point?

File: stdpopsim/utils.py [hunk 2]
```diff
@@ -13,3 +13,32 @@ def is_valid_demographic_model_id(model_id):
...
+    regex = re.compile(r"[A-Z][a-z]+ [a-z]+")
+    return regex.fullmatch(name) is not None
+
+
+def is_valid_species_common_name(common_name):
+    """
+    Returns True if the specified string is a valid species common name. This
+    must start with a capital letter.
+    """
+    # FIXME any sensible restrictions we can make on common names? See #330.
```
Comment (andrewkern) [inline stdpopsim/utils.py]: saw this and looked at https://github.com/popgensims/stdpopsim/issues/330. 

gosh i don't think we can put a restriction on common name. there are things out there with weirdo names

File: stdpopsim/utils.py [hunk 3]
```diff
@@ -13,3 +13,32 @@ def is_valid_demographic_model_id(model_id):
...
+    regex = re.compile(r"[A-Z][a-z]{,2}[A-Z][a-z]{,2}")
+    return regex.fullmatch(species_id) is not None
+
+
+def is_valid_species_name(name):
+    """
+    Returns True if the specified string is a valid species name. This
+    must be two or more words with first letter capitalised.
+    """
+    # FIXME this only supports two words for now. See #329
```
Comment (andrewkern) [inline stdpopsim/utils.py]: left a note on https://github.com/popgensims/stdpopsim/issues/329

File: tests/test_genetic_maps.py [hunk 4]
```diff
@@ -54,10 +54,11 @@ class GeneticMapTestClass(genetic_maps.GeneticMap):
     def __init__(self):
         genome = stdpopsim.Genome(chromosomes=[])
         _species = stdpopsim.Species(
-            id="tesspe", name="Test species", genome=genome)
+            id="TesSpe", name="Test species", common_name="Testy McTestface",
```
Comment (andrewkern) [inline tests/test_genetic_maps.py]: !!!

File: stdpopsim/catalog/arabidopsis_thaliana.py (hunk 1 above)
Comment (jeromekelleher) [reply stdpopsim/catalog/arabidopsis_thaliana.py]: "Salome map" is pretty undescriptive..

**Comment by jeromekelleher** [comment]
Thanks @andrewkern, going to merge.
